cmd/node/node_test.go

```diff
@@ -426,49 +426,28 @@ func TestSpacemeshApp_GrpcService(t *testing.T) {
 	r.NoError(err)
 	r.Equal(false, app.Config.API.StartNodeService)
 
-	const message = "Hello World"
-
-	var conn *grpc.ClientConn
-	require.Eventually(t, func() bool {
-		var err error
-		conn, err = grpc.Dial(fmt.Sprintf("localhost:%d", port), grpc.WithTransportCredentials(insecure.NewCredentials()))
-		return err == nil
-	}, 2*time.Second, 100*time.Millisecond)
-
-	r.NoError(err)
-	c := pb.NewNodeServiceClient(conn)
-
-	// We expect this one to fail
-	_, err = c.Echo(context.Background(), &pb.EchoRequest{
-		Msg: &pb.SimpleString{Value: message},
-	})
-	r.Error(err)
-	r.Contains(err.Error(), "rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing dial tcp")
-	r.NoError(conn.Close())
+	conn, err := grpc.Dial(fmt.Sprintf("localhost:%d", port), grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithBlock(), grpc.WithTimeout(3*time.Second))
+	r.ErrorContains(err, "context deadline exceeded")
 
 	resetFlags()
 	events.CloseEventReporter()
 
-	// Test starting the server from the commandline
+	// Test starting the server from the command line
 	// uses Cmd.Run from above
 	str, err = testArgs("--grpc-port", strconv.Itoa(port), "--grpc", "node")
 	r.Empty(str)
 	r.NoError(err)
 	r.Equal(port, app.Config.API.GrpcServerPort)
-	r.Equal(true, app.Config.API.StartNodeService)
-
-	require.Eventually(t, func() bool {
-		var err error
-		conn, err = grpc.Dial(fmt.Sprintf("localhost:%d", port), grpc.WithTransportCredentials(insecure.NewCredentials()))
-		return err == nil
-	}, 2*time.Second, 100*time.Millisecond)
-	defer conn.Close()
+	r.True(app.Config.API.StartNodeService)
 
+	conn, err = grpc.Dial(fmt.Sprintf("localhost:%d", port), grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithBlock())
```

Comment:
What do you plan for handling timeouts (`grpc.Dial()` here will wait for connection indefinetely)? 

I suggest to use `grpc.WithTimeout(time.Second*5)` option.

Reply:
I'll add timeouts to the Dials with the same values as the `r.Eventually` used before.